src/CLRunner.ts

```diff
@@ -1358,8 +1358,10 @@ export class CLRunner {
                         entityId: entityId,
                         values: []
                     } as CLM.FilledEntity
-                    filledEntityMap.map[entity.entityName] = filledEntity
                     filledEntityMap.map[entity.entityId] = filledEntity
+                    if (bidirectional) {
```

Comment:
Why does it need to have both mappings some times but not others?

There are like 3 different methods for extending a map by name to also have map by id, or to convert to only have id, etc and it's hard to know which to use.  I thought it would be simpler to always use a single map and know it has a mapping of whatever have available name or id.  If this isn't possible to due cases like above I wondered why?

Reply:
Would take some investigation/refactoring to clean up.  The renderers need a mapping in both directions.  CreateFilledEntityMap expands the uni-directional map (which is what the traindialog has) to a bi-directional map before calling the renderers.  We could potentially pass the renderers two separate maps and have two distinct map types.  Would make it easier to understand

Reply:
Yea, there is also `convertToMapById` and `addEntitiesById` ( Think only add is used since it covers all cases of convert) The TakeTextAction and TakeAPIAction etc used `addEntitiesById` but doesn't look like they do anymore